api/xds_protocol.rst

```diff
@@ -155,45 +231,109 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
 to any :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` with a :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` containing the
 ACK/NACK immediately after it has been either accepted or rejected. If
 the management server provides the same set of resources rather than
-waiting for a change to occur, it will cause Envoy and the management
-server to spin and have a severe performance impact.
+waiting for a change to occur, it will cause needless work on both the client and the management
+server, which could have a severe performance impact.
 
 Within a stream, new :ref:`DiscoveryRequests <envoy_api_msg_DiscoveryRequest>` supersede any prior
 :ref:`DiscoveryRequests <envoy_api_msg_DiscoveryRequest>` having the same resource type. This means that
 the management server only needs to respond to the latest
 :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` on each stream for any given resource type.
 
-Resource hints
-^^^^^^^^^^^^^^
+How the client specifies what resources to return
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+xDS requests allow the client to specify a set of resource names as a hint to the server about which resources the client
+is interested in. In the SotW API variants, this is done via the :ref:`resource_names
+<envoy_api_field_DiscoveryRequest.resource_names>` specified in the :ref:`DiscoveryRequest
+<envoy_api_msg_DiscoveryRequest>`; in the incremental API variants, this is done via the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>` and
+:ref:`resource_names_unsubscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_unsubscribe>` fields in the
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`.
 
-The :ref:`resource_names <envoy_api_field_DiscoveryRequest.resource_names>` specified in the :ref:`DiscoveryRequest
-<envoy_api_msg_DiscoveryRequest>` are a hint about what resources the client is interested in.
+For `Listener` and `Cluster` resource types, Envoy will always set the resource hints to empty, in which case the server
+should use site-specific business logic to determine the full set of resources that the client is interested in, typically
+based on the client's :ref:`node <envoy_api_msg_Core.Node>` identification. However, other xDS clients (such as gRPC) may
+specify explicit LDS/CDS resources as resource hints, for example if they only have a singleton listener and already know
```

Comment:
Can you pick a name for the gRPC client that distinguishes it from other uses of gRPC in this doc and use that throughout? Otherwise the reader may be confused between the gRPC client implementing xDS with the use of gRPC by the xDS protocol.